Source/Controllers/DataControllers/SPTableData.m

```diff
@@ -442,75 +442,89 @@ - (BOOL) updateInformationForCurrentView
-
-	// Ensure the encoding is set to UTF8
-	if (changeEncoding) {
-		[mySQLConnection storeEncodingForRestoration];
-		[mySQLConnection setEncoding:@"utf8mb4"];
-	}
-
-	// In cases where this method is called directly instead of via -updateInformationForCurrentTable
-	// (for example, from the exporters) clear the list of constraints to prevent the previous call's table
-	// constraints being included in the table information (issue 1206).
-	[constraints removeAllObjects];
+    BOOL changeEncoding = ![[mySQLConnection encoding] hasPrefix:@"utf8"];
 
-	// Retrieve the CREATE TABLE syntax for the table
-	SPMySQLResult *theResult = [mySQLConnection queryString:[NSString stringWithFormat:@"SHOW CREATE TABLE %@", [tableName backtickQuotedString]]];
-	[theResult setReturnDataAsStrings:YES];
+    // Catch unselected tables and return nil
+    if ([tableName isEqualToString:@""] || !tableName) return nil;
 
-	// Check for any errors, but only display them if a connection still exists
-	if ([mySQLConnection queryErrored]) {
-		if ([mySQLConnection isConnected]) {
-			NSString *errorMessage = [NSString stringWithFormat:NSLocalizedString(@"An error occurred while retrieving the information for table '%@'. Please try again.\n\nMySQL said: %@", @"error retrieving table information informative message"),
-					   tableName, [mySQLConnection lastErrorMessage]];
-
-			// If the current table doesn't exist anymore reload table list
-			if ([mySQLConnection lastErrorID] == 1146) {
-
-				// Release the table loading lock to allow reselection/reloading to requery the database.
-				pthread_mutex_unlock(&dataProcessingLock);
-
-				[tableListInstance deselectAllTables];
-				[tableListInstance updateTables:self];
-			}
-			SPMainQSync(^{
-				[NSAlert createWarningAlertWithTitle:NSLocalizedString(@"Error retrieving table information", @"error retrieving table information message") message:errorMessage callback:nil];
-			});
-			if (changeEncoding) [mySQLConnection restoreStoredEncoding];
-		}
+    // Ensure the encoding is set to UTF8
+    if (changeEncoding) {
+        [mySQLConnection storeEncodingForRestoration];
+        [mySQLConnection setEncoding:@"utf8mb4"];
+    }
 
-		return nil;
-	}
+    // In cases where this method is called directly instead of via -updateInformationForCurrentTable
+    // (for example, from the exporters) clear the list of constraints to prevent the previous call's table
+    // constraints being included in the table information (issue 1206).
+    [constraints removeAllObjects];
+
+    // Retrieve the CREATE TABLE syntax for the table
+    SPMySQLResult *theResult;
+    if (database)
+        theResult = [mySQLConnection queryString:[NSString stringWithFormat:@"SHOW CREATE TABLE %@.%@", [database backtickQuotedString], [tableName backtickQuotedString]]];
+    else
+        theResult = [mySQLConnection queryString:[NSString stringWithFormat:@"SHOW CREATE TABLE %@", [tableName backtickQuotedString]]];
```

Comment:
What if instead of the if block we just always specify the db and fallback on the current db if db isn't specified? Probably cleaner than the if block. Suggestion, not a required change but I think it would be cleaner

Reply:
Yes, that's what I tried first, but sometimes [tableDocumentInstance database] return nil, for example when using CSV export. And I did not find another way to retrieve the default value.
But if you know how, I'm interested!